### DISCUSSION TITLE
How to debug raw Guzzle / curl request?

### DISCUSSION THREAD

**Discussion by boryn** [saloonphp/saloon#226] [General]
I wonder how we could debug the raw curl query sent by Guzzle?

I tried with:
```php
protected function defaultConfig(): array
{
    return [
        'debug' => true,
    ];
}
```

but actually where we can see the debugging output?

**Comment by Sammyjo20** [comment]
Hey @boryn my favourite way is using a stream to create a temporary file somewhere to write to...

```php
protected function defaultConfig(): array
{
    return [
        'debug' => fopen('some-file.txt', 'wb+'),
    ];
}
```

When using Laravel, I use `base_path('temp/debug.txt`)` or something like that
